SDKBuildScripts/Update-PlayFabSdk.ps1

```diff
@@ -0,0 +1,160 @@
+[CmdletBinding(SupportsShouldProcess = $true, DefaultParameterSetName = "ApiSpecGitUrl")]
+param(
+    [Parameter(Position = 1)]
+    [ValidateSet(
+        "ActionScriptSDK",
+        "Cocos2d-xSDK",
+        "CSharpSDK",
+        "JavaSDK",
+        "JavaScriptSDK",
+        "LuaSDK",
+        "NodeSDK",
+        "Objective_C_SDK",
+        "PhpSDK",
+        "PostmanCollection",
+        "PythonSDK",
+        "SdkTestingCloudScript",
+        "UnrealMarketplacePlugin",
+        "UnitySDK",
+        "WindowsSDK",
+        "XPlatCppSDK")]
+    [string[]]$SdkNames,
+    [Parameter(ParameterSetName="ApiSpecPath")]
+    [string]$ApiSpecPath,
+    [Parameter(ParameterSetName="ApiSpecPfUrl")]
+    [string]$ApiSpecPfUrl,
+    [Parameter(ParameterSetName="ApiSpecPfUrlBranch")]
+    [string]$ApiSpecBranch,
+    [Parameter(ParameterSetName="ApiSpecGitUrl")]
+    [switch]$ApiSpecGitUrl,
+    [switch]$KeepSource,
+    [switch]$NonNullable,
+    [switch]$Beta
+)
+
+begin
+{
+    if(!(Get-Command node))
+    {
+        throw "You must have Node.js installed to generate the SDK"
+    }
+
+    $sdkTargetSrcMap = @{
+        "ActionScriptSDK" = "actionscript";
+        "Cocos2d-xSDK" = "cpp-cocos2dx";
+        "CSharpSDK" = "csharp";
+        "JavaSDK" = "java";
+        "JavaScriptSDK" = "javascript";
+        "LuaSDK" = "LuaSdk";
+        "NodeSDK" = "js-node";
+        "Objective_C_SDK" = "objc";
+        "PhpSDK" = "PhpSdk";
+        "PostmanCollection" = "postman";
+        "PythonSDK" = "PythonSdk";
+        "SdkTestingCloudScript" = "SdkTestingCloudScript";
+        "UnrealMarketplacePlugin" = "UnrealMarketplacePlugin";
+        "UnitySDK" = "unity-v2";
+        "WindowsSDK" = "windowssdk";
+        "XPlatCppSDK" = "xplatcppsdk";
+    }
+
+    $sdksPath = Resolve-Path (Join-Path $PSScriptRoot "..\..\sdks")
+    if(!(Test-Path $sdksPath))
+    {
+        mkdir $sdksPath
+    }
+
+    if($ApiSpecBranch)
```

Comment:
The better condition might be: If ApiSpecPfUrl is not already defined...

Reply:
ApiSpecUrl and ApiSpecCloud/Vertical are defined in separate parameter sets, so it's actually impossible to pass BOTH parameters at the same time so there's no need to check that.